torch/_dynamo/variables/builder.py

```diff
@@ -1117,10 +1129,9 @@ def wrap_tensor(self, value: torch.Tensor):
         # TODO: I think the result is guaranteed to be fake with
         # ignore_subclass changes
         # Note: this information is conveyed via subclass_type now
-        fake_tensor_value = None
-        example_value = tensor_variable.proxy.node.meta["example_value"]
-        if is_fake(example_value):
-            fake_tensor_value = example_value
+        fake_tensor_value = tensor_variable.proxy.node.meta["example_value"]
```

Comment:
This may not belong in this PR?

Further, I think this may not always be true? (example value may not always be fake even if tensor).

The only situations in which we instantiate an `example_value` manually (rather than running `get_fake_value`) are:
1. `wrap_unspecialized_primitive`
2. `wrap_numpy_ndarray`
3. Wrapping `SymNodeVariable`s and `TensorVariable`s in Higher order op: `validate_args_and_maybe_create_graph_inputs` under the `manually_set_subgraph_inputs` setting (when is this used?)
4. Basically every higher order op creates a new example value (but we try to reduce the duplicate FakeTensor generation by using previously cached `example_value`s

Reply:
Generally, I think we can benefit from guaranteeing that `wrap_fx_proxy`, when wrapping a tensor-like, will always return a `FakeTensor` wrapped by the current output graph / instance of Dynamo.

Reply:
The other corner case we have to deal with is that when we try to wrap a tensor value via `VariableBuilder`, if the value is not in `self.tx.output.side_effects`, we may actually double wrap the tensor.

This codepath is the one run for `wrap_numpy_ndarray` and `var_getattr` on a `nn.Module`.

Let me try to ensure that this is never the case.